what4/src/What4/Expr/Builder.hs

```diff
@@ -590,12 +590,12 @@ data App (e :: BaseType -> Type) (tp :: BaseType) where
         -> !(e (BaseBVType w))   -- rotate amount
         -> App e (BaseBVType w)
 
-  BVZext :: (1 <= w, w+1 <= r, 1 <= r)
```

Comment:
I'd prefer we not change the syntax for BVZext and BVSext; that is, still require that they be non-trivial extensions.  Since we are doing a test for the noop case in `bvSext` and `bvZext` operations anyway, the syntax should only be applied in non-trivial cases anyway.

In other code (e.g., SMTWriter) we rely on the extension amount being nonzero, so it's better to preserve that invariant, I think.